### DISCUSSION TITLE
Inconsistent result with place_id from `/search` in `/details`

### DISCUSSION THREAD

**Discussion by foodev** [osm-search/Nominatim#2628] [Q&A]
We have a custom nominatim server with only some countries imported. Sometimes we get a 404 when we use the `place_id` from a `/search` result to query `/details`.

What is going on here? I would expect that if `/search` gives me a result, `/details` does too. Is it because we have not imported all the countries? But then why does `/search` give a result?

Example:

`/search.php?format=json&postalcode=03179&countrycodes=ua&addressdetails=1`

0 |  
-- | --
place_id | 139413610
licence | "Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright"
boundingbox |  
0 | "50.297595716667"
1 | "50.617595716667"
2 | "30.184193233192"
3 | "30.504193233192"
lat | "50.4575957166667"
lon | "30.3441932331917"
display_name | "03179"
class | "place"
type | "postcode"
importance | 0.335
address |  
postcode | "03179"
country_code | "ua"

But `/details?place_id=139413610&addressdetails=1&format=json` (the `place_id` from the result above):

error |  
-- | --
code | 404
message | "No place with that place ID found."

---

When I run the same queries on https://nominatim.openstreetmap.org I do get results (of course I updated the second query to use the "place_id" from the result of the first query), but "/details" gives me a different "place_id" in the results than the one I was looking for.

https://nominatim.openstreetmap.org/search.php?format=json&postalcode=03179&countrycodes=ua&addressdetails=1

vs.

https://nominatim.openstreetmap.org/details?place_id=286268327&addressdetails=1&format=json

Is that because of the `postcode` type?

**Comment by lonvia** [comment]
Nominatim misses a details view for couple of "special" places, among them artificial postcodes. Issue #1671 has the full list of concerned objects. As a rule of thumb: if the results doesn't have a osm_type then there won't be a detailed view. Usually Nominatim would return information about the 'parent object' for these places when asking for details. That's what the OSM instance does. I suspect that in your instance the postcode is an outlier that does not have a parent place and so Nominatim gives up. Is the Ukraine one of the countries that you have imported?

**Reply by foodev** [reply]
Yes, we have imported Ukraine. In fact, all of Europe: https://download.geofabrik.de/europe.html

**Reply by lonvia** [reply]
Can you please check, if your postcodes have been indexed at all: `SELECT count(*) FROM location_postcode WHERE indexed_status > 0` should return 0. If that is not the case, something might have gone wrong during import.

**Reply by foodev** [reply]
It returns `0`...

**Reply by foodev** [reply]
I also don't understand why `address` on `nominatim.openstreetmap.org` contains data for `borough` and `country`, but the result from our server does not.

**Reply by lonvia** [reply]
Can you find "Sviatoshynskyi district" and "Kyiv" in your local installation and in what country are they reported to be?

**Reply by foodev** [reply]
They both have `country_code=ua`.

**Reply by lonvia** [reply]
No idea what went wrong then. How many postcodes are there with parent_place_id = 0? You can force a reindex on them by running the SQL `UPDATE location_postcode SET indexed_status = 2 WHERE parent_place_id = 0` and then running the command `nominatim index`.

**Reply by foodev** [reply]
```
nominatim=# SELECT COUNT(*) FROM location_postcode WHERE parent_place_id = 0;
 count 
-------
   788
(1 row)
```
Then we might try a reindex... thanks anyway.
